src/xdplwf/offload.c

```diff
@@ -6,6 +6,13 @@
 #include "precomp.h"
 #include "offload.tmh"
 
+//
+// Annotate routines that must be invoked from the serialized offload work
+// queue.
+//
+#define _Offload_work_routine_
+#define _Requires_offload_rundown_ref_
```

Comment:
We might be able to leverage the lock annotations under the cover to get real help from SAL on these annotations.

Reply:
Opened https://github.com/microsoft/xdp-for-windows/issues/218 to track this.

Reply:
Agreed.